### PR TITLE
Fixed #24904 -- Fixed nested namespaces in current_app.

### PR DESCRIPTION
**Pull request by knbk**
Fixed reverse() to correctly handled nested namespace lookups
in current_app.

https://code.djangoproject.com/ticket/24904

### CODE REVIEW

File: tests/urlpatterns_reverse/tests.py [hunk 1]
```diff
@@ -574,6 +574,25 @@ def test_namespaces_with_variables(self):
         self.assertEqual('/inc70/', reverse('inc-ns5:inner-nothing', args=['70']))
         self.assertEqual('/inc78/extra/foobar/', reverse('inc-ns5:inner-extra', args=['78', 'foobar']))
 
+    def test_nested_app_lookup(self):
+        "A nested current_app should be split in individual namespaces: see #24904"
```
Comment (timgraham) [inline tests/urlpatterns_reverse/tests.py]: namespaces (#24904).

File: tests/urlpatterns_reverse/tests.py [hunk 2]
```diff
@@ -574,6 +574,25 @@ def test_namespaces_with_variables(self):
...
+        self.assertEqual('/ns-included1/test4/inner/', reverse('inc-ns1:testapp:urlobject-view'))
+        self.assertEqual('/ns-included1/test4/inner/37/42/', reverse('inc-ns1:testapp:urlobject-view', args=[37, 42]))
```
Comment (timgraham) [inline tests/urlpatterns_reverse/tests.py]: could you please use this style so lines don't exceed 119 characters? 
```
self.assertEqual(
    '/ns-included1/test4/inner/37/42/',
    reverse('...')
)

File: docs/releases/1.8.3.txt [hunk 3]
```diff
@@ -50,3 +50,6 @@ Bugfixes
 
 * Fixed recording of applied status for squashed (replacement) migrations
   (:ticket:`24628`).
+
+* Fixed :func:`~django.core.urlresolvers.reverse()` not handling nested
```
Comment (timgraham) [inline docs/releases/1.8.3.txt]: I think this bug has been around forever and thus doesn't qualify for a backport. Did I miss some justification?

File: tests/urlpatterns_reverse/tests.py (hunk 2 above)
Comment (knbk) [inline tests/urlpatterns_reverse/tests.py]: Are the coding style guidelines preferred over local consistency? I actually prefer the vertical alignment with the url on the left and reverse() on the right, which makes it much more comprehensible in my opinion. I can still make that fit the 119 character limit.

Comment (timgraham) [inline tests/urlpatterns_reverse/tests.py]: I won't complain as long as your choice makes sense and is within 119 chars.

**Comment by timgraham** [comment]
For the tests, I really much prefer the indentation I suggested. Indenting a non-multiple of 4 number of spaces and cramming everything on the right side is not very readable in my opinion.

**Comment by timgraham** [comment]
merged in f0450c9b1298bf5f39e229aec76c9bf7d57d4479, thanks!

**Pull request closed without merging**